libraries/chain/db_maint.cpp

```diff
@@ -72,12 +72,45 @@ vector<std::reference_wrapper<const typename Index::object_type>> database::sort
    return refs;
 }
 
-template<class... Types>
-void database::perform_account_maintenance(std::tuple<Types...> helpers)
+template<class Type>
+void database::perform_account_maintenance(Type tally_helper)
 {
-   const auto& idx = get_index_type<account_index>().indices().get<by_name>();
-   for( const account_object& a : idx )
-      detail::for_each(helpers, a, detail::gen_seq<sizeof...(Types)>());
+   const auto& bal_idx = get_index_type< account_balance_index >().indices().get< by_maintenance_flag >();
+   if( bal_idx.begin() != bal_idx.end() )
+   {
+      auto bal_itr = bal_idx.rbegin();
+      while( bal_itr->maintenance_flag )
```

Comment:
Must check bal_itr != bal_idx.rend(), then you can remove enclosing if.

Reply:
Oh, now I see - if bal_idx is not empty() then bal_idx.rbegin() will always exist and the while loop ends when all entries with maintenance_flag have been handled. Ok.